hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/ClusteringPlanActionExecutor.java

```diff
@@ -98,6 +98,7 @@ public Option<HoodieClusteringPlan> execute() {
             .setExtraMetadata(extraMetadata.orElse(Collections.emptyMap()))
             .setClusteringPlan(planOption.get())
             .build();
+        table.validateForLatestTimestamp(clusteringInstant.getTimestamp());
```

Comment:
I'm wondering if we can get rid of such per-action handling which is hard to maintain.  Instead, could we design a general approach to maintain instant time ordering?

Reply:
+1

Reply:
We discussed that Hudi 1.0 solves the problem overall with True Time API and completion.  For Hudi 0.x, asynchronous table service scheduling should be guarded by the lock to avoid such an issue described in the PR.  We're going to release 1.0.0 release soon so as a low-effort change (a general approach takes a large effort and ROI is low compared to releasing 1.0.0), this PR with validations suffices to guard the cases in Hudi 0.x where the table service scheduling is not guarded by the lock.

Reply:
synced up w/ Ethan f2f. We do not want to invest more into 0.x for now. anyways 1.0 guarantees monotonically increasing commit times. So for now, we will go ahead w/ this patch as is. may not take up non-trivial refactoring

Reply:
ditto: retry instead of throws.